### ISSUE TITLE
Issue during inference and prediciton of model probabilites from release version 0.3.2.

### ISSUE THREAD

**Issue by Akshay1-6180** [qubvel/segmentation_models.pytorch#746]
So currently I have lightning 2.0.1 and torch 2.0 installed and when I upgrade to v0.3.2 and then during the model training and validation its fine but after the model weights are saved and I run prediction probabilities then all the probabilities turn out to be almost the same and it gives poor AUROC , i don't have this issue with releases less than v0.3.2 , moreover during the training of lightning , i get this error of broken pipe with the newer versions but the training still occurs
![image](https://user-images.githubusercontent.com/65217827/229872846-6411d471-3b3b-4835-8574-9d0cc519887f.png)

**Comment by cschloer** [comment]
Hi - I'm having a similar issue. But specifically related to batch size. So inference works fine if I use the same batch size as training, but when I input a single image I got awful performance (I am using eval mode to handle the BatchNorm2D).

Is this also what you are experiencing?

**Comment by Akshay1-6180** [comment]
I havent checked the batch size one , but have the same issue with the single image u mentioned , regarding batchnorm 2D , i use model.eval so it should disable maybe there is some bug in the new version. The problem is the saved model weights dont seem to mimic the weights of the trained model but a random model , have attached an example of the probabilities that are generated in v0.3.2 with the target , notice how all are the same, didnt have this issue with other version
![image](https://user-images.githubusercontent.com/65217827/229902248-57950d5a-9c43-48a6-8ce8-ceafc154aa72.png)

**Comment by Akshay1-6180** [comment]
> 

yep similar

**Comment by qubvel** [comment]
What was the previous release you don't have any issues with?

**Comment by Akshay1-6180** [comment]
0.3.1
